aem-authoring-toolkit-plugin/src/main/java/com/exadel/aem/toolkit/core/handlers/widget/WidgetContainerHandler.java

```diff
@@ -0,0 +1,74 @@
+        String defaultTabName = annotation.equals(TabsWidget.class) ? ContainerHandler.TAB : ContainerHandler.ACCORDION;
+        String exceptionMessage = annotation.equals(TabsWidget.class) ? ContainerHandler.TABS_EXCEPTION : ContainerHandler.ACCORDION_EXCEPTON;
+        Element tabItemsElement = (Element) element
+                .appendChild(getXmlUtil().createNodeElement(DialogConstants.NN_ITEMS));
+
+        Map<String, TabContainerInstance> tabInstancesFromCurrentClass = getInstancesFromCurrentClass(annotation, field);
+
+        Class<?> tabsType = field.getType();
+        List<Field> allFields = WidgetSetHandler.getContainerFields(element, field, tabsType);
+
+        if (tabInstancesFromCurrentClass.isEmpty() && !allFields.isEmpty()) {
+            PluginRuntime.context().getExceptionHandler().handle(new InvalidSettingException(
+                    exceptionMessage + tabsType.getName()
+            ));
+            if (annotation.equals(TabsWidget.class)) {
+                tabInstancesFromCurrentClass.put(StringUtils.EMPTY, new TabContainerInstance("newTab"));
+            }
+        }
+
+        ContainerHandler.addTabs(tabInstancesFromCurrentClass, allFields, ArrayUtils.EMPTY_STRING_ARRAY, tabItemsElement, defaultTabName);
+        ContainerHandler.handleInvalidTabException(allFields);
+    }
+
+    /**
+     * Get all tabs from current class.
+     *
+     * @param annotation {@link Class<? extends Annotation>} searching annotation
+     * @param field      Current {@code Field} instance
+     */
+    private Map<String, TabContainerInstance> getInstancesFromCurrentClass(Class<? extends Annotation> annotation, Field field) {
+        Map<String, TabContainerInstance> tabInstancesFromCurrentClass = new LinkedHashMap<>();
+        if (field.isAnnotationPresent(annotation)) {
+            if (annotation.equals(TabsWidget.class)) {
+                Arrays.stream(field.getAnnotation(TabsWidget.class).tabs())
+                        .forEach(tab -> {
+                            TabContainerInstance tabContainerInstance = new TabContainerInstance(tab.title());
+                            tabContainerInstance.setAttributes(ContainerHandler.getAnnotationFields(tab));
+                            tabInstancesFromCurrentClass.put(tab.title(), tabContainerInstance);
+                        });
+            } else if (annotation.equals(AccordionWidget.class)) {
+                Arrays.stream(field.getAnnotation(AccordionWidget.class).panels())
+                        .forEach(tab -> {
+                            TabContainerInstance tabContainerInstance = new TabContainerInstance(tab.title());
+                            tabContainerInstance.setAttributes(ContainerHandler.getAnnotationFields(tab));
+                            tabInstancesFromCurrentClass.put(tab.title(), tabContainerInstance);
+                        });
+            }
+        }
```

Comment:
this code seems to be similar to https://github.com/exadel-inc/aem-authoring-toolkit/pull/93/files#diff-4dfc292c1d2ed88aa1e08d2c6e268e5a22a3556339974cf026c452325d9985b9R170-R184
can we somehow reduce the duplication?